### PR TITLE
[SG2] Add transform node UI

### PR DESCRIPTION
**Pull request by dhsavell**
### Purpose of this PR

This PR implements a UI for the transform node, which was implemented previously. Now the source/destination spaces and transform type can be changed.

image

---
### Testing status

- Manually compared to SG1.

---
### Comments to reviewers

As with the last node UI PR, there are some new CLDS usages that eventually need to be changed.

When testing, note that the preview is always 2D.

### CODE REVIEW

**Review by bencloward — APPROVED** [review]
Looks and functions really well!

**Comment by Nightmask3** [comment]
image

The searcher preview for this node seems off for whatever reason

File: com.unity.sg2/Editor/GraphUI/EditorCommon/CommandStateObserver/NodeUICommands.cs [hunk 1]
```diff
@@ -108,38 +108,56 @@ class SetGradientTypeValueCommand : UndoableCommand
         }
     }
 
-    class SetSwizzleMaskCommand : UndoableCommand
+    abstract class SetNodeFieldCommand<T> : UndoableCommand
```
[the remark this answers is not in the document]
Comment (Nightmask3) [reply com.unity.sg2/Editor/GraphUI/EditorCommon/CommandStateObserver/NodeUICommands.cs:111]: Maybe if we were to have a marshalling function in the `ShaderGraphModel` that was used to actually communicate with the CLDSModel and set any fields, and then we had some Enumerations/other identifiers that mapped to those field types and value setters in the `ShaderGraphModel`?

That way any view can call a command as such `SetNodeFieldCommand<CLDSFieldsEnumeration.ConversionType>();`
and then we respond to that appropriately, calling the actual field setter within `ShaderGraphModel`.

**Comment by bencloward** [comment]
> image
> 
> The searcher preview for this node seems off for whatever reason

Unfortunately, this is true for all nodes right now.  So it's not just this one.

File: com.unity.sg2/Editor/GraphUI/GraphElements/ModelUI/TransformDropdownsPart.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                };
+            }
+
+            m_Root = new VisualElement();
+
+            GraphElementHelper.LoadTemplateAndStylesheet(m_Root, "TransformDropdownsPart", "sg-transform-dropdowns");
+
+            m_FromDropdown = m_Root.Q<EnumField>("from");
+            m_FromDropdown.Init(default(CoordinateSpace));
+            m_FromDropdown.RegisterValueChangedCallback(SetCoordinateSpaceCallback(GraphDelta.TransformNode.kSourceSpace));
```
Comment (Nightmask3) [inline com.unity.sg2/Editor/GraphUI/GraphElements/ModelUI/TransformDropdownsPart.cs:40]: Something I learned from the memory leak stuff I fixed in SG1 is that we need to have an explicit step in our application lifecycle to release resources and unbind callbacks, which is something to think about as we add any callbacks going forwards

Not a blocker for this PR, as I think we need a more general purpose solution, but wanted to put that on your radar

**Review by Nightmask3 — APPROVED** [review]

**Pull request merged**